R/get_vpts_aloft.R

```diff
@@ -0,0 +1,122 @@
+#' Get VPTS data from aloftdata
+#'
+#' This function retrieves VPTS (Vertical Profile Time Series) data from the
+#' aloftdata server.
+#'
+#' - Constructs the S3 paths for the VPTS files based on the input
+#' - Performs parallel HTTP requests to fetch the vpts-csv data
+#' - Parses the response bodies with some assumptions about the column classes
+#' - Adds a column with the radar source
+#' - Overwrites the radar column with the radar_odim_code, all other values for
+#' this column are considered in error.
+#'
+#'
+#' @param radar_odim_code The radar ODIM code.
+#' @param rounded_interval The interval to fetch data for, rounded to nearest
+#'   day
+#' @param source The source of the data. One of baltrad, uva or ecog-04003.
+#' @param coverage A data.frame containing the coverage of the aloft data
+#'   repository. If not provided, it will be fetched from via the internet.
+#' @return A list of vpts data.frames
+#'
+#' @importFrom dplyr .data
+#' @importFrom lubridate %within%
+#'
+#' @examples
+#' get_vpts_aloft(
+#'   "bejab",
+#'   lubridate::interval("20240305", "20240307"),
+#'   "baltrad"
+#' )
+#'
+#' @noRd
+get_vpts_aloft <- function(radar_odim_code,
+                           rounded_interval,
+                           source,
+                           coverage = aloft_data_coverage()) {
+  # rename source argument for readability
+  selected_source <- source
+
+  # Check that only one radar is provided (string of length 1)
+  if (!rlang::is_string(radar_odim_code)) {
+    cli::cli_abort(
+      "Please provide (only one) radar as a character vector of length 1.",
+      class = "getRad_error_radar_not_single_string"
+    )
+  }
+
+  # Check that radar_odim_code is a single 5 character string
+  if (nchar(radar_odim_code) != 5) {
+    cli::cli_abort(
+      "Radar ODIM code must be a single 5 character string.",
+      class = "getRad_error_radar_odim_code_invalid"
+    )
+  }
+
+  # Check if the requested radars are present in the coverage
+  if (!all(radar_odim_code %in% coverage$radar)) {
+    cli::cli_abort(
+      "Radar(s) not found in ALOFT coverage:
+      {radar_odim_code[!radar_odim_code %in% coverage$radar]}.",
+      class = "getRad_error_aloft_radar_not_found"
+    )
+  }
+
+  # Check if the requested date radar combination is present in the coverage
+  filtered_coverage <- dplyr::filter(
+    coverage,
+    .data$radar %in% radar_odim_code,
+    .data$date %within% rounded_interval,
+    .data$source == selected_source
+  )
+  at_least_one_radar_date_combination_exists <- nrow(filtered_coverage) > 0
+
+  if (!at_least_one_radar_date_combination_exists) {
+    cli::cli_abort(
```

Comment:
In some circumstances this might be an annoying check. Say you are doing an analysis per year for all radars you found with `weather_radars()`. Without first doing a manual check against `aloft_data_coverage` this error will trigger. Maybe it is better to do as an warning? lets discuss

Reply:
Discussed: 
- drop radar date combinations that do not exist
- warn for combinations that do not.
- Return the rest.